src/realm/object-store/c_api/error.cpp

```diff
@@ -297,3 +311,12 @@ RLM_EXPORT bool realm_wrap_exceptions(void (*func)()) noexcept
         return true;
     });
 }
+
+RLM_API void realm_register_user_code_callback_error(void* usercode_error) noexcept
+{
+    realm::c_api::wrap_err([=]() {
```

Comment:
A bit convoluted way to store the pointer. It could just be stored directly in the thread local ErrorStorage.